.github/workflows/verify.yaml

```diff
@@ -0,0 +1,94 @@
+name: Verify
+
+on:
+  push:
+    branches:
+      - main
+      - "release-*"
+      - "test/**"
+  pull_request:
+    branches:
+      - main
+      - "release-*"
+
+env:
+  GO_VERSION: "1.22.1"
+
+jobs:
+  verify-deps:
+    name: Dependencies
+    runs-on: ubuntu-24.04
+    timeout-minutes: 5
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - run: |
+          make verify-godeps
+
+  verify-formatting:
+    name: Formatting
+    runs-on: ubuntu-24.04
+    timeout-minutes: 5
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```

Comment:
We really do need it in two places: git-diff --check white-space error checking (technically wouldn't need to fetch all, just enough to get base_ref), and for verifying/generating docs. For binary building it could be dropped, it would just result in a different built version than what you'd get locally (since git describe would produce a different result).

Reply:
Thanks for the clarification, @klihub!